examples/c++/bmi160.cxx

```diff
@@ -41,10 +42,30 @@ void sig_handler(int signo)
 int main(int argc, char **argv)
 {
   signal(SIGINT, sig_handler);
+  upm::BMI160 *sensor = NULL;
 //! [Interesting]
 
-  // Instantiate a BMI160 instance using default i2c bus and address
-  upm::BMI160 *sensor = new upm::BMI160();
+  // Instantiate a BMI160 instance using default i2c bus and address.
+  // If that fails, try default spi bus.
+  try
+  {
+    sensor = new upm::BMI160();
+  }
+  catch (std::exception& e)
+  {
+    std::cout << "Failed to connect via I2C, trying default SPI bus" << std::endl;
+  }
+
+  try
+  {
+    if (sensor == NULL)
+      sensor = new upm::BMI160(0, -1);
```

Comment:
This is unlikely to work without specifying the correct CS pin.  The default is -1 (as 3rd arg), which is suitable for a device in which the OS/kernel handles CS for you (like Edison on pin 10).  But on other platforms, you would need to actually specify whatever pin you connected CS to, which can be anything.